sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -60,9 +68,31 @@ def test_affinity_propagation():
     assert np.unique(labels).size == n_clusters_
     assert n_clusters == n_clusters_
 
-    # Test also with no copy
+
+def test_affinity_propagation_no_copy():
+    """
+    Check behaviour of not copying the input data.
+    """
+    S = -euclidean_distances(X, squared=True)
+    S_original = S.copy()
+    preference = np.median(S) * 10
+    assert not np.allclose(S.diagonal(), preference)
+
+    # once copy=True S should not be modified
```

Comment:
```suggestion
    # with copy=True S should not be modified
```
An english language nitpick. "once" makes me think of ordering in time ("once the sun sets, the vampires come out"), which isn't really what is happening here. I think.

(same for the comment below)